subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformationChain.java

```diff
@@ -25,10 +25,34 @@
 
     private final Transformation first;
     private final Transformation second;
+    private final int stepsCount;
 
     public TransformationChain(Transformation first, Transformation second) {
         this.first = first;
         this.second = second;
+        this.stepsCount = first.stepsCount() + second.stepsCount();
+    }
+
+    @Override
+    public boolean contains(Transformation otherTransform) {
```

Comment:
Maybe `contains` should be `endsWith`.

Reply:
Indeed, renamed it three times during dev, but that sounds better than what I tried